src/org/way2it/data_structures/list/List.java

```diff
@@ -14,30 +16,61 @@ public List() {
     // Should add new value to the end of the list, increment length and extend array by x2 if needed
     public void add(String value) {
         // TODO implement me
+        if (length == values.length) {
+            String[] temp = values;
+            this.values = new String[values.length+1];
+            System.arraycopy(temp,0,values, 0, temp.length);
+        }
+        values[length] = value;
+        length++;
     }
 
     // Should add new value at the specified index, moving other values to the right,
     // increment length and extend array by x2 if needed
     public void addAtIndex(String value, int index) {
         // TODO implement me
+        if (length == values.length) {
+            String[] temp = values;
+            this.values = new String[values.length+1];
+            System.arraycopy(temp,0,values, 0, temp.length);
+        }
+        for (int i = values.length-2; i >= index; i--) {
```

Comment:
It would be enough to move values starting from `length + 1`, not from `values.length - 2`